azure_functions_worker/bindings/meta.py

```diff
@@ -184,3 +210,41 @@ def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
         return protos.ParameterBinding(
             name=out_name,
             data=rpc_val)
+
+
+def get_deferred_binding(bind_name: str,
+                         pytype: typing.Optional[type] = None) -> object:
+    binding = None
+
+    # Checks if pytype is a supported sdk type
+    if (SDK_BINDING_REGISTRY is not None
+            and SDK_BINDING_REGISTRY.check_supported_type(pytype)):
+        # Set flag once
+        global DEFERRED_BINDINGS_ENABLED
+        if not DEFERRED_BINDINGS_ENABLED:
+            DEFERRED_BINDINGS_ENABLED = True
+        binding = SDK_BINDING_REGISTRY.get(bind_name)
+
+    # This will return None if not a supported type
+    return binding
+
+
+def deferred_bindings_decode(binding: typing.Any,
+                             pb: protos.ParameterBinding, *,
+                             pytype: typing.Optional[type],
+                             datum: typing.Any,
+                             metadata: typing.Any):
+    global SDK_CACHE
+    # Check is the object is already in the cache
+    # If dict is empty or key doesn't exist, obj is None
+    obj = SDK_CACHE.get((pb.name, pytype, datum.value.content), None)
```

Comment:
todo: rename from obj. explain what should be returned. rename everything sdk into deferred_bindings.